src/fields/field.impl.hh

```diff
@@ -60,6 +60,19 @@ Field<spacedim,Value>::Field(const string &name, bool bc)
 
 
 template<int spacedim, class Value>
+Field<spacedim,Value>::Field(unsigned int component_index)
+: data_(std::make_shared<SharedData>())
```

Comment:
The name and input_name should also be passed as parameters of this constructor. This way we
1. get rid of setting input_name explicitly in setup_components
2. better document that input_name is set to the name of the multifield.